src/python/pants/goal/anonymous_telemetry.py

```diff
@@ -0,0 +1,184 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+import json
+import logging
+import re
+import uuid
+from typing import cast
+
+from humbug.consent import HumbugConsent  # type: ignore
+from humbug.report import Report, Reporter  # type: ignore
+
+from pants.engine.internals.scheduler import Workunit
+from pants.engine.rules import collect_rules, rule
+from pants.engine.streaming_workunit_handler import (
+    StreamingWorkunitContext,
+    WorkunitsCallback,
+    WorkunitsCallbackFactory,
+    WorkunitsCallbackFactoryRequest,
+)
+from pants.engine.unions import UnionRule
+from pants.option.subsystem import Subsystem
+
+logger = logging.getLogger(__name__)
+
+
+_bugout_access_token = "3ae76900-9a68-4a87-a127-7c9f179d7272"
+_bugout_journal_id = "801e9b3c-6b03-40a7-870f-5b25d326da66"
+_telemetry_docs_url = "https://www.pantsbuild.org/docs/anonymous-telemetry"
+_telemetry_docs_referral = f"See {_telemetry_docs_url} for details"
+
+
+class AnonymousTelemetry(Subsystem):
+    options_scope = "anonymous-telemetry"
+    help = (
+        "Options related to sending anonymous stats to the Pants project, to aid development. "
+        "Repos that opt into this telemetry will be able to access to their own stats."
+    )
+
+    @classmethod
+    def register_options(cls, register):
+        register(
+            "--enabled",
+            advanced=True,
+            type=bool,
+            help=(
+                f"Whether to send anonymous telemetry to the Pants project. Telemetry is sent "
+                f"asynchronously, with silent failure, and does not impact build times or "
+                f"outcomes. {_telemetry_docs_referral}."
+            ),
+        )
+        register(
+            "--repo-id",
+            advanced=True,
+            type=str,
+            help=(
+                f"An anonymized ID representing this repo.  For private repos, you likely want the "
```

Comment:
I think the linter will complain about f-string syntax being used when no actual interpolation happens.